File: Desafio_Tecnico_Cadastro_de_Beneficiarios/Domain/Entities/BeneficiarioModel.cs
using Desafio_Tecnico_Cadastro_de_Beneficiarios.Domain.Enums;

namespace Desafio_Tecnico_Cadastro_de_Beneficiarios.Domain.Entities
{
    public class BeneficiarioModel
    {
        public int Id { get; set; }
        public string NomeCompleto { get; set; }
        public string Cpf { get; set; }
        public DateTime DataNascimento { get; set; }

        public DateTime DataCadastro { get; set; } = DateTime.UtcNow;
        public Status Status { get; set; } = Status.ATIVO;

        public int PlanoId { get; set; }

        public PlanoModel Plano { get; set; }
    }

}

File: frontend/src/app/models/beneficiario.model.ts
export interface Beneficiario {
  id: number;
  nomeCompleto: string;
  cpf: string;
  email: string;
  dataNascimento: string;
  planoId: number;
  plano?: {
    id: number;
    nome: string;
  };
}

export interface BeneficiarioCriacao {
  nomeCompleto: string;
  cpf: string;
  email: string;
  dataNascimento: string;
  planoId: number;
}
